src/utils/makeReactNativeConfig.js

```diff
@@ -92,7 +92,8 @@ const getDefaultConfig = ({
         { parser: { requireEnsure: false } },
         {
           test: /\.js$/,
-          exclude: /node_modules\/(?!react|@expo|pretty-format|haul|metro)/,
+          // eslint-disable-next-line no-useless-escape
+          exclude: /node_modules(?!.*[\/\\](react|@expo|pretty-format|haul|metro))/,
```

Comment:
Similar to the `babel-register` `ignore` option, this can use a `function` instead:
```js
exclude: function (filename) {
    const normalizedPath = filename.replace('\\', '/');
    const indexOfNodeModules = normalizedPath.lastIndexOf('/node_modules/');

    for (const safeModule of ['react', '@expo', 'pretty-format', 'haul', 'metro']) {
        const indexOfSafeModule = normalizedPath.lastIndexOf(`/node_modules/${safeModule}/`);
        if (indexOfSafeModule > -1 && indexOfSafeModule >= indexOfNodeModules) {
            return false;
        }
    }

    if (indexOfNodeModules > -1) {
        return true;
    }

    return false;
}
```

That will probably out-perform the Regex, be more clear, and be more reliable.

Reply:
> That will probably out-perform the Regex

Not sure if we need perf gains in this area, regexps are pretty fast already, what's slow is Babel compilation

Reply:
The risk right now is that the Regex will miss
`node_modules/random/haul23`
which is probably not desirable.